### DISCUSSION TITLE
Redirecting within S3

### DISCUSSION THREAD

**Discussion by billygcode** [aws/aws-sdk-go-v2#1988] [Q&A]
If I have an application with multiple custom S3 compatible endpoints (server1, server2, server3), and I create a s3 client with one of the endpoints (server1). 

If the client(server1) makes a call, let's say PutObject(), and the endpoint server(server1) determines that the request should be in a different endpoint location(server3) and returns a 301/302/307/308 with a Redirect Location(server3) in the header.

Does the SDK automatically redirect the 30X to that endpoint(server3) or would I have to implement my own redirecting?

**Comment by RanVaknin** [comment]
Hi @billygcode ,
307 and 308 status code would redirect automatically. 
For other codes you'll have to implement your own behavior.

Thanks,
Ran

**Reply by billygcode** [reply]
It seems that all of the requests that I send are capable of redirecting 307 and 308 except those with a body (putObject).

Am I correct in assuming that the SDK creates a http.Request using smithy-go( https://github.com/aws/smithy-go/blob/v1.13.5/transport/http/request.go#L161 ) and then proceeds to send the request using the standard Golang library (https://github.com/golang/go/blob/master/src/net/http/client.go)?
